api/v1beta1/kafkacluster_types.go

```diff
@@ -850,6 +854,11 @@ func (eConfig *EnvoyConfig) GetTopologySpreadConstaints() []corev1.TopologySprea
 	return eConfig.TopologySpreadConstraints
 }
 
+// GetPodSecurityContext returns the security context for the envoy deployment podspec.
+func (eConfig *EnvoyConfig) GetPodSecurityContext() *corev1.PodSecurityContext {
+	return eConfig.PodSecurityContext
```

Comment:
I can see there is no check for that elsewhere but we can use here:
if eConfig == nil {
return nil
}